tools/build_langserver/langserver/analyzer.go

```diff
@@ -369,14 +376,103 @@ func (a *Analyzer) BuildLabelFromContent(ctx context.Context,
 		return nil
 	}
 
-	if item := asp.WalkAST(stmts, callback); item != nil {
+	if item := asp.WalkAST(val, callback); item != nil {
 		return item.(*BuildLabel)
 	}
 
 	return nil
 }
 
-// VariablesFromContent returns a map of variable name to Variable
+// GetSubinclude returns a Subinclude object based on the statement and uri passed in.
+func (a *Analyzer) GetSubinclude(ctx context.Context, stmts []*asp.Statement, uri lsp.DocumentURI) map[string]*RuleDef {
+
+	ruleDefs := make(map[string]*RuleDef)
+
+	currentPkg, err := PackageLabelFromURI(uri)
+	if err != nil {
+		log.Warning("fail to load package from uri %s: %s", uri, err)
+	}
+	for _, stmt := range stmts {
+		if stmt.Ident != nil {
+			ident := a.identFromStatement(stmt)
+			if ident.Type == "call" && ident.Name == "subinclude" && len(ident.Action.Call.Arguments) > 0 {
+				includeLabel := ident.Action.Call.Arguments[0].Value.Val.String
```

Comment:
this can panic if the argument is not a string... e.g. 
```
something = "//build_defs:whatever"
subinclude(something)
```
which is technically legal, if unusual. Best check that `.Value.Val` is not nil.

Reply:
Not sure if I can panic here, because that would actually shut down the LPS. however I could log it and highlight it if `.Value.Val` is nil

Reply:
yes that's what I meant, you shouldn't panic, but the previous implementation might have :)